### ISSUE TITLE
createFacade type error with interface

### ISSUE THREAD

**Issue by Aaronkala** [garbles/react-facade#2]
`createFacade` errors when being used with the interface type. [Typescript recommends interfaces over intersections](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections).

```tsx
type Type = { useX: () => void };
interface Interface { useX: () => void };
createFacade<Type>;
createFacade<Interface>; // Type 'Interface' does not satisfy the constraint 'FacadeInterface'. Index signature for type 'string' is missing in type 'Interface'.
```

[Codesandbox](https://codesandbox.io/s/autumn-sun-lm9ymu?file=/src/index.ts)

Maybe the type could be improved, or is the only way around using `type` instead of `interface`? I am using [this rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-definitions.md) to enforce the use of interfaces over types so it will require an exception to the rule :)

Great library btw! Looking forward to start using it more

**Comment by garbles** [comment]
Hey, that's definitely unintentional.

I can take a look at it in a few days, but feel free to open a PR (with a noop test to verify its working) if you've got a working solution now.

Edit: I found a solution, so I'll push that once I've cleaned it up.

**Comment by garbles** [comment]
@Aaronkala just published `0.3.1` and this should be fixed.

**Comment by Aaronkala** [comment]
Wow that was fast. Thanks!
